src/main/cpp/domconfigurator.cpp

```diff
@@ -1035,6 +1037,19 @@ void DOMConfigurator::parse(
 		MessageBufferUseStaticStream();
 	}
 
+	LogString threadSignalValue = subst(getAttribute(utf8Decoder, element, THREAD_TYPE_ATTR));
+	if( !threadSignalValue.empty() && threadSignalValue != NuLL ){
```

Comment:
While it's not strictly related to your PR, `NuLL` reads a bit surprising to me. How about changig that to something like `[STR_]NULL[_STR]` instead?

Reply:
yeah that's probably a good idea - it looked odd to me too at first, and then I realized it is to avoid conflicts with the NULL macro.